Code/Framework/AzFramework/AzFramework/TargetManagement/TargetManagementComponent.cpp

```diff
@@ -299,6 +299,7 @@ namespace AzFramework
         if (target_autoconnect && !m_isTargetHost && !m_targetJoinThread->IsRunning() &&
             m_networkInterface->GetConnectionSet().GetActiveConnectionCount() == 0)
         {
+            m_targetJoinThread->Join();
```

Comment:
Why would a thread join before it starts?

Reply:
The first instance of the Join would be a no-op given as you'd think. Once the thread successfully establishes a connection, it Stops. If that connection terminates (say by closing the Lua IDE), we get back into this logic in which case it'd need to Join before Starting.